pkg/cmd/roachtest/roachtestutil/mixedversion/mixedversion.go

```diff
@@ -836,14 +876,90 @@ func latestPredecessor(_ *rand.Rand, v *clusterupgrade.Version) (*clusterupgrade
 }
 
 // randomPredecessor is an implementation of `predecessorFunc` that
-// picks a random predecessor for the given release version.
-func randomPredecessor(rng *rand.Rand, v *clusterupgrade.Version) (*clusterupgrade.Version, error) {
+// picks a random predecessor for the given release version. If we are
+// choosing a predecessor in the same series as the minimum supported
+// version, special care is taken to select a random predecessor that
+// is more recent that the minimum supported version.
+func randomPredecessor(
+	rng *rand.Rand, v, minSupported *clusterupgrade.Version,
+) (*clusterupgrade.Version, error) {
 	predecessor, err := release.RandomPredecessor(rng, &v.Version)
 	if err != nil {
 		return nil, err
 	}
 
-	return clusterupgrade.MustParseVersion(predecessor), nil
+	// If the minimum supported version is from a different release
+	// series, we can pick any random patch release.
+	predV := clusterupgrade.MustParseVersion(predecessor)
+	if predV.Series() != minSupported.Series() {
+		return predV, nil
+	}
+
+	// If the latest release of a series is a pre-release, we validate
+	// whether the minimum supported version is valid.
+	if predV.PreRelease() != "" && !predV.AtLeast(minSupported) {
```

Comment:
What if the latest release is a pre-release *and* is supported? The comment below:

> At this point, we know that the latest predecessor is not a pre-release

Implies that we should also have a return case for that here?

Reply:
Although it looks like the `Patch()` of pre releases is 0 so it will return `predV` anyway. So maybe just tweaking the comment is fine if we don't care to have an explicit return case for it.

Reply:
Yes, I thought about this case but figured it would be too specific to mention. Someone would need to write a test where the minimum supported version is a release that is barely out of the oven (i.e., a beta/rc). Very unlikely. But I'll try to rephrase the comment.